Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -476,7 +477,7 @@ ThrowCompletionOr<Value> Value::to_number(GlobalObject& global_object) const
     case Type::Double:
         return *this;
     case Type::String: {
-        auto string = as_string().string().trim_whitespace();
+        String string = Utf8View(as_string().string()).trim(whitespace_characters, AK::TrimMode::Both).as_string();
```

Comment:
The fact that we allocate a `String` in `Value::to_number()` is super sad. Can we use `StringView` all the way here instead?

Reply:
Not really, strtod expects a null terminated string